sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -235,6 +254,8 @@ def __init__(self, fully_qualified_namespace, eventhub_name, credential, **kwarg
         self._container_id = CONTAINER_PREFIX + str(uuid.uuid4())[:8]
         if isinstance(credential, AzureSasCredential):
             self._credential = AzureSasTokenCredential(credential)
+        elif isinstance(credential, AzureNamedKeyCredential):
+            self._credential = AzureNamedKeyTokenCredential(credential) # type: ignore
```

Comment:
dumb question: why do we have this `# type: ignore`? -- I didn't see one for `self._credential = AzureSasTokenCredential(credential)`)

Reply:
yes we need it unfortunately -  we assign a different type (intentionally) to the same variable- so we'd get this error
```
Incompatible types in assignment (expression has type "EventhubAzureNamedKeyTokenCredential", variable has type "EventhubAzureSasTokenCredential")
```